src/runtime-rs/crates/hypervisor/src/qemu/network.rs

```diff
@@ -0,0 +1,372 @@
+    // "vfio-ap"
+    VfioAp,
+}
+
+impl ToString for NetDeviceWithTransport {
+    fn to_string(&self) -> String {
+        match self {
+            NetDeviceWithTransport::VirtioNetPci => "virtio-net-pci".to_owned(),
+            NetDeviceWithTransport::VirtioNetCcw => "virtio-net-ccw".to_owned(),
+            NetDeviceWithTransport::VhostUserNet => "virtio-net".to_owned(),
+            NetDeviceWithTransport::VfioPci => "vfio-pci".to_owned(),
+            NetDeviceWithTransport::VfioCcw => "vfio-ccw".to_owned(),
+            NetDeviceWithTransport::VfioAp => "vfio-ap".to_owned(),
+        }
+    }
+}
+
+impl TryFrom<&str> for NetDeviceWithTransport {
+    type Error = anyhow::Error;
+
+    // method to parse a PciPath from a string
+    fn try_from(s: &str) -> Result<Self> {
+        let transport = match s {
+            "virtio-net-pci" => NetDeviceWithTransport::VirtioNetPci,
+            "virtio-net-ccw" => NetDeviceWithTransport::VirtioNetCcw,
+            "virtio-net" => NetDeviceWithTransport::VhostUserNet,
+            "vfio-pci" => NetDeviceWithTransport::VfioPci,
+            "vfio-ccw" => NetDeviceWithTransport::VfioCcw,
+            "vfio-ap" => NetDeviceWithTransport::VfioAp,
+            _ => return Err(anyhow!("unsupported transport")),
+        };
+
+        Ok(transport)
+    }
+}
+
+#[derive(Debug, Default, PartialEq)]
+enum NetDeviceDriver {
+    #[default]
+    VirtioNet,
+    Vhostuser,
+    Vfio,
+}
+
+impl ToString for NetDeviceDriver {
+    fn to_string(&self) -> String {
+        match self {
+            NetDeviceDriver::VirtioNet => "virtio-net".to_owned(),
+            NetDeviceDriver::Vhostuser => "vhost-user".to_owned(),
+            NetDeviceDriver::Vfio => "vfio".to_owned(),
+        }
+    }
+}
+
+#[derive(Debug, Default, PartialEq)]
+enum NetDeviceType {
+    /// Tap is a tap networking device type.
+    /// TAP NetDeviceType = "tap"
+    #[default]
+    Tap,
+    /// MacTap is a macvtap networking device type.
+    /// MACVTAP NetDeviceType = "macvtap"
+    MacvTap,
+    /// IpTap is a ipvtap virtual networking device type.
+    /// IPVTAP NetDeviceType = "ipvtap"
+    IpvTap,
+    /// VethTap is a veth-tap virtual networking device type.
+    /// VETHTAP NetDeviceType = "vethtap"
+    VethTap,
+    /// VhuNet is a vhost-user port (socket)
+    /// VhuNet NetDeviceType = "vhostuser"
+    VhuNet,
+    /// Physical is a direct assigned PCI device or PCI VF
+    /// Physical NetDeviceType = "VFIO"
+    Physical,
+}
+
+impl NetDeviceType {
+    // QemuNetdevParam converts to the QEMU -netdev parameter notation
+    fn qemu_netdev_param(&self, transport: VirtioTransport) -> Option<&str> {
+        match self {
+            // -netdev type=tap -device virtio-net-pci
+            NetDeviceType::VethTap
+            | NetDeviceType::Tap
+            | NetDeviceType::MacvTap
+            | NetDeviceType::IpvTap => Some("tap"),
+            // -netdev type=vhost-user (no device)
```

Comment:
This comment you borrowed from govmm looks wrong. It certainly doesn't make sense to pass a  `-netdev` host backend without a companion guest visible `-device`...